### DISCUSSION TITLE
Where to put the file after download

### DISCUSSION THREAD

**Discussion by lennyvinnierhodes** [Geolykt/EnchantmentsPlus#136] [Q&A]
i dont know where to put the file after i have downloaded it

**Comment by Geolykt** [comment]
A) Which file did you download?
B) What server are you running? (e.g. forge, fabric, quilt, sponge, spigot, paper, etc.)

Generally it should be in the `plugins` folder. But since you are asking this question the chance is relatively high that that folder does not exist and that you are not running a server that is verified to be working with this plugin

**Reply by lennyvinnierhodes** [reply]
A) [EnchantmentsPlus-4xx.zip](https://github.com/Geolykt/EnchantmentsPlus/files/10148178/EnchantmentsPlus-4xx.zip)

B) im running forge 1.12.2

**Reply by Geolykt** [reply]
A) That is the source zip/tarball. You'd need to compile it by decompressing it and running maven on it (i.e. `mvn package` after installing maven)
B.1) The 4XX dev builds do not run on the Java version required by forge 1.12. Java 16 or later is required by EnchantmentsPlus. Forge breaks with Java 11 or later last time I checked.
B.2) EnchantmentsPlus does not work on 1.12 as it has been updated to newer versions and as such it will not understand a few things (zombified piglins are the prime example. The flattening is a lesser known one, albeit very important)
B.3) EnchantmentsPlus should run on Bukkit-compliant server software, vanilla forge is definitely one of them and even hybrids will have their compliance issues.
